### DISCUSSION TITLE
Desktop open graph (Java 8)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#177] [Q&A] [answered]
Thanks for the test. I continue to struggle with the last install. But still keep losing that battle though.
[GH offline graph failure video](https://youtu.be/bDlhkiAqxr8url)

_Originally posted by @0709wiwiwi in https://github.com/devemux86/cruiser/discussions/166#discussioncomment-5975747_

**Comment by devemux86** [comment]
@0709wiwiwi in the video you seem to stay on the parent folder of belgium.

On desktop, open a folder means to **enter** the folder and press open:

![Cruiser](https://github.com/devemux86/cruiser/assets/3484020/85ea0a37-40e1-4fb7-af3f-2237f08b6720)

**Reply by 0709wiwiwi** [reply]
[Old version_old graph](https://youtu.be/BAXqFgci_CY)

**Reply by devemux86** [reply]
- Video 2: you correctly enter the graph folder and press open
- Video 1: you do not enter the graph folder
You remain in parent folder and press open

**Reply by 0709wiwiwi** [reply]
[New version_new graph](https://youtu.be/WqqyM20IMU8)
No matter how I click I can't get it right.

**Reply by devemux86** [reply]
In this video you seem to enter the graph folder and then press open.
This should work, unless the graph folder does not contain valid data.

Are you sure you are running the Cruiser 3.0.20?
Are you sure the graph folder contains valid data?

If the folder does not contain routing data,
there is immediately a warning message after press open.

**Reply by 0709wiwiwi** [reply]
> _Are you sure you are running the Cruiser 3.0.20?_ 

 yes checked
> _Are you sure the graph folder contains valid data?_ 

yes checked. 
![Valid graph](https://github.com/devemux86/cruiser/assets/54167201/861567f2-d9e9-4f79-a7e5-3a9efeea95d6)

>_If the folder does not contain routing data,
there is immediately a warning message after press open._ 

No warning.

**Comment by devemux86** [comment]
@0709wiwiwi 

Select the "Settings | App theme | **Default**", restart Cruiser and try again.

Maybe the file browser of Java's default theme works better on your Windows.

**Reply by 0709wiwiwi** [reply]
> Select the "Settings | App theme | **Default**", restart Cruiser and try again.

No change.

Ok let's stop for now.  Time to get wiser.

**Comment by slwmr** [comment]
I had the same problem. You have to upgrade your Java Environment. The Cruiser 3.0.20 (or graphhopper to be more precise) is using access methods not available in older Java Runtime Environments. 
Exception in thread "Thread-9" java.lang.NoSuchMethodError: java.nio.ByteBuffer.position(I)Ljava/nio/ByteBuffer;
at com.graphhopper.storage.MMapDataAccess.getBytes(MMapDataAccess.java:427).

**Reply by devemux86** [reply]
@slwmr thanks, very nice!

@0709wiwiwi @slwmr what Java do you use?

I can add a compiler parameter to fix this in next update.
At least Java 11 should work properly?

**Reply by slwmr** [reply]
I had this error with Java 8. When I changed the start script to point to OpenJDK12 the offline routing with GH started to work without any problems. 
Thank you for a great application. I'm also using it to show the results of some simple programming of my son.  So maybe you contribute to the next generation of developers in this way ;) .

**Reply by 0709wiwiwi** [reply]
![image](https://github.com/devemux86/cruiser/assets/54167201/ec5b8ead-f118-4da2-9c69-cf7ed156edb9)

**Comment by devemux86** [comment]
@0709wiwiwi @slwmr 

Can you try on **Java 8** the desktop Cruiser 3.0.21 (Beta) available here?

https://www.dropbox.com/sh/itq3e3d0wr0zp3q/AADZ0BYypUf91GhMVKryZzmba

**Reply by 0709wiwiwi** [reply]
Result  is OK !

**Comment by slwmr** [comment]
I have tested the beta version with Java 8 as well. The offline routing is working!

**Comment by devemux86** [comment]
@0709wiwiwi @slwmr thanks for the report and for finding the Java error!

The fix will be included in the next desktop Cruiser update.

**Comment by devemux86** [comment, marked as answer]
It should work properly with [desktop Cruiser 3.0.21](https://github.com/devemux86/cruiser/releases).
